### ISSUE TITLE
TextFiled keyboardType missing UIKeyboardTypeASCIICapable for iOS

### ISSUE THREAD

**Issue by tgcasce** [flutter/flutter#127831]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Use case

iOS has UIKeyboardTypeASCIICapable, but Flutter missing this type.

typedef NS_ENUM(NSInteger, UIKeyboardType) {
    UIKeyboardTypeDefault,                // Default type for the current input method.
    UIKeyboardTypeASCIICapable,           // Displays a keyboard which can enter ASCII characters
    UIKeyboardTypeNumbersAndPunctuation,  // Numbers and assorted punctuation.
    UIKeyboardTypeURL,                    // A type optimized for URL entry (shows . / .com prominently).
    UIKeyboardTypeNumberPad,              // A number pad with locale-appropriate digits (0-9, ۰-۹, ०-९, etc.). Suitable for PIN entry.
    UIKeyboardTypePhonePad,               // A phone pad (1-9, *, 0, #, with letters under the numbers).
    UIKeyboardTypeNamePhonePad,           // A type optimized for entering a person's name or phone number.
    UIKeyboardTypeEmailAddress,           // A type optimized for multiple email address entry (shows space @ . prominently).
    UIKeyboardTypeDecimalPad API_AVAILABLE(ios(4.1)),   // A number pad with a decimal point.
    UIKeyboardTypeTwitter API_AVAILABLE(ios(5.0)),      // A type optimized for twitter text entry (easy access to @ #)
    UIKeyboardTypeWebSearch API_AVAILABLE(ios(7.0)),    // A default keyboard type with URL-oriented addition (shows space . prominently).
    UIKeyboardTypeASCIICapableNumberPad API_AVAILABLE(ios(10.0)), // A number pad (0-9) that will always be ASCII digits.

    UIKeyboardTypeAlphabet = UIKeyboardTypeASCIICapable, // Deprecated

};

### Proposal

maybe Flutter can add this type

class TextInputType {
static const TextInputType asciiCapble = TextInputType._(11);
}

**Comment by huycozy** [comment]
Related to a feature request at #107970 (see https://github.com/flutter/flutter/issues/107970#issuecomment-1322815434)

**Comment by justinmc** [comment]
Good idea, I think we should probably support this keyboard type.  I think this would require two small PRs, one in the engine and one in the framework.

**Comment by letrungdo** [comment]
Merged in PR: https://github.com/flutter/engine/pull/50293/files
=> Flutter 3.22.0 supported
use TextInputType.visiblePassword

**Comment by jmagman** [comment]
Twitter keyboard implemented with https://github.com/flutter/flutter/pull/161025
